golang/cli_test/cli_test.go

```diff
@@ -0,0 +1,419 @@
+package clitest
+
+import (
+	"testing"
+	"encoding/hex"
+	"github.com/stretchr/testify/suite"
+)
+
+type UtilsSuite struct {
+	suite.Suite
+	TestData TestData
+}
+
+// Runs the whole test suite
+func TestRelay(t *testing.T) {
+
+	utilsSuite := new(UtilsSuite)
+	utilsSuite.TestData = GrabTestData(t)
+
+	suite.Run(t, utilsSuite)
+}
+
+func (suite *UtilsSuite) TestRelayCLIIsAncestor() {
+	suite.T().Parallel()
+
+	genesisHeaders := suite.TestData.GenesisHeaders
+	newDiffHeaders := suite.TestData.NewDiffHeaders
+
+	// Initialize CHain
+	f := InitFixtures(suite.T())
+	proc := f.RelayDStart()
+	defer proc.Stop(false)
```

Comment:
these are breaking the linter because we don't check for error conditions. Is it possible to check errors of deferred operations or should I have the linter ignore these?

Reply:
the recomended parttern is to make a helper function that does this:

```
  f := InitFixtures(suite.T())
  proc := f.RelayDStart()

  defer func() {
    err := f.Close(); 
    if err != nil {
      /// do whatever
    }
  }()
```